features/step_definitions/tag_steps.rb

```diff
@@ -79,15 +79,9 @@
   rel.add_association(fand)
 end
 
-Given /^a canonical (\w+) "([^\"]*)"$/ do |tag_type, tagname|
+Given /^a (non-?c|c)anonical (\w+) "([^\"]*)"$/ do |canonical, tag_type, tagname|
```

Comment:
Thanks for combining the two similar step definitions! However, we had a bit of trouble parsing this. Could we rework the variable names to be a little clearer on the first read? 

```
Given /^a (noncanonical|canonical) (\w+) "(.*?)"$/ do |canonical_status, tag_type, tag_name|
  tag = tag_type.classify.constantize.find_or_create_by_name(tag_name)
  tag.canonical = canonical_status == "canonical"
  tag.save
end
```

(I have no idea why we used `tagname` as the variable name elsewhere in this file, since we generally prefer _not_ to squish words together and instead use underscores, but things happen!)

Reply:
The variable naming suggestions make total sense.

I was trying to be clever with the first regex and allow use of any among "canonical", "noncanonical", or "non-canonical" (with optional hyphen).  Was that too much?  The hyphenated version wasn't previously supported, and to be fair I probably only put that in because it got tripped up by it.

Reply:
No, the hyphen part was fine! The confusing bit was just the part where it didn't use the whole word to determine the canonical status. If it'll work and you want to do `(non-?canonical|canonical)`, you're more than welcome to go for it!